assignment/assignment-impl/impl/src/java/org/sakaiproject/assignment/impl/BaseAssignmentService.java

```diff
@@ -10518,13 +10896,13 @@ private String getReviewReport(ContentResource cr)
 		}
 		
 		//TODO: delete this and all calling methods if there are no repercussions
-		private ContentResource getFirstAcceptableAttachement() {
+		private ContentResource getFirstAcceptableAttachement(boolean allowAnyFile) {
 			String contentId = null;
 			try {
 			for( int i =0; i < m_submittedAttachments.size();i++ ) {
 				Reference ref = (Reference)m_submittedAttachments.get(i);
 				ContentResource contentResource = (ContentResource)ref.getEntity();
-				if (contentReviewService.isAcceptableContent(contentResource)) {
+				if (contentReviewService.isAcceptableSize(contentResource) && (allowAnyFile || contentReviewService.isAcceptableContent(contentResource))){
```

Comment:
The intention of contentReviewService.isAcceptableContent(ContentResource) was to handle all of these checks in one spot. isAcceptableSize and allowAnyFile should be refactored into CRS.isAcceptableContent()